Add vanilla shovel rightclick ability to Tinkers shovels + excavators

KnightMiner opened the pull request:
Allows shovels to do the vanilla shovel feature of making grass paths. It will use AOE.

bonii-xx commented:
Still needs a check if the tool is broken, also looking at the mattock again, this should be easier: 
https://github.com/SlimeKnights/TinkersConstruct/blob/master/src/main/java/slimeknights/tconstruct/tools/item/Mattock.java#L108-L127

However this does not allow you to rightclick on path to extend it. Hardcode that part in the onItemUse directly. This way we stay more flexible and compatible.

KnightMiner commented:
Fixed. I was able to use almost the same code as the mattocks actually, I did not think about them doing AOE (the expanders are a little new). Also squashed.
